moved/src/state_actor.rs

```diff
@@ -1188,4 +1234,157 @@ mod tests {
+        let initial_balance = U256::from(15);
+        let amount = U256::from(4);
+        let (mut state_actor, _) = create_state_actor_with_fake_queries(
+            EVM_ADDRESS.to_move_address(),
+            initial_balance,
+            StatePayloadId,
+        );
+
+        let signer = Signer::new(&PRIVATE_KEY);
+        let mut tx = TxEip1559 {
+            chain_id: CHAIN_ID,
+            nonce: signer.nonce,
+            gas_limit: u64::MAX,
+            max_fee_per_gas: 0,
+            max_priority_fee_per_gas: 0,
+            to: TxKind::Call(to),
+            value: amount,
+            access_list: Default::default(),
+            input: Default::default(),
+        };
+        let signature = signer.inner.sign_transaction_sync(&mut tx).unwrap();
+        let tx = TxEnvelope::Eip1559(tx.into_signed(signature));
+
+        state_actor.handle_command(Command::AddTransaction { tx });
+
+        let (tx, rx) = oneshot::channel();
+
+        state_actor.handle_command(Command::StartBlockBuild {
+            payload_attributes: Default::default(),
+            response_channel: tx,
+        });
+
+        let payload_id = rx.blocking_recv().unwrap();
+        let (tx, rx) = oneshot::channel();
+
+        state_actor.handle_query(Query::GetPayload {
+            id: payload_id,
+            response_channel: tx,
+        });
+
+        let expected_payload = rx.blocking_recv().unwrap();
+
+        let mut tx = TxEip1559 {
+            chain_id: CHAIN_ID,
+            nonce: signer.nonce + 1,
+            gas_limit: u64::MAX,
+            max_fee_per_gas: 0,
+            max_priority_fee_per_gas: 0,
+            to: TxKind::Call(to),
+            value: amount,
+            access_list: Default::default(),
+            input: Default::default(),
+        };
+        let signature = signer.inner.sign_transaction_sync(&mut tx).unwrap();
+        let tx = TxEnvelope::Eip1559(tx.into_signed(signature));
+
+        state_actor.handle_command(Command::AddTransaction { tx });
+
+        let (tx, rx) = oneshot::channel();
+
+        state_actor.handle_command(Command::StartBlockBuild {
+            payload_attributes: Payload {
+                timestamp: U64::from(1u64),
+                ..Default::default()
+            },
+            response_channel: tx,
+        });
+
+        let payload_2_id = rx.blocking_recv().unwrap();
+        let (tx, _rx) = oneshot::channel();
+
+        assert_ne!(
+            payload_id, payload_2_id,
+            "Payload IDs of different blocks should not match"
+        );
+
+        state_actor.handle_query(Query::GetPayload {
+            id: payload_2_id,
+            response_channel: tx,
+        });
+
+        let (tx, rx) = oneshot::channel();
+
+        state_actor.handle_query(Query::GetPayload {
+            id: payload_id,
+            response_channel: tx,
+        });
+
+        let actual_payload = rx.blocking_recv().unwrap();
```

Comment:
Instead, we can call it `old_payload` or `same_payload`

Reply:
True, I considered it, but ultimately I still have a reason to stick with this naming as stated in the other comment.